src/puppetlabs/trapperkeeper/services/webserver/jetty9_core.clj

```diff
@@ -228,7 +229,9 @@
                            ssl-config)
                         :client-auth :none
                         :cipher-suites (or (:cipher-suites ssl-config) config/acceptable-ciphers)
-                        :protocols     (or (:protocols ssl-config) config/default-protocols)}))
+                        :protocols     (or (:protocols ssl-config) config/default-protocols)
+                        :allow-renegotiation  (or (:allow-renegotiation ssl-config)
```

Comment:
This has the same issue as [above](https://github.com/puppetlabs/trapperkeeper-webserver-jetty9/pull/186/files#r144926581). I'm curious as to why we're setting this in the config and here? If you know off the top of your head I'd love to hear it, otherwise I'll look deeper after lunch.

Reply:
About the ability to over ride the default (disabling renegotiations). I think the latest update should allow it to enable renegotiations by adjusting it in config file. At least I was able to in my testing and it seems to work as intended but I too am curious to know what case could be missed.  
About setting it in config and above, I just mimicked what is being done for other configuration settings like protocols, cipher-suites etc.   But if you think there is some redundancy I can experiment and figure the minimal change set to achieve the desired functionality.

Reply:
I dove into this a bit and I think that we should update [ProxSslConfig](https://github.com/puppetlabs/trapperkeeper-webserver-jetty9/pull/186/files/8c457c5941862d6af036dafa73f374903497853b#diff-1fc7ef9acd5fbc52947b0a4ba167e727R82) above to include an optional key for `:allow-renegotiations`. I also think we should look at making that key required in [`config/WebserverSslContextFactory`](https://github.com/puppetlabs/trapperkeeper-webserver-jetty9/pull/186/files/8c457c5941862d6af036dafa73f374903497853b#diff-83ec7cd15abf3ccb0d9023ae974afc20R174).

I can go into why I think that, but I spent the last while writing a tome that I think might actually just obscure the actionable part.

Reply:
I have left the new key optional in config/WebserverSslContextFactory. One reason being it would require specifying it in all config files of the different projects using TK-webserver-jetty9 and just fear that someone might set it to true. Or else I would need to change all the default config files for the repo's using TK. 
Instead given the motivation for introducing this configuration parameter I would prefer leaving it disabled by default and not require it to be able also specified in the config. If someone (really) wants it enabled they can. 
But would like to understand better why making it required may be good idea.

Reply:
So, IIRC, making it a required key in the two schemas that I mentioned would not make it required in an actual config file. To do that we would need to make it required in the `WebserverRawConfig`.

The two schemas I've mentioned define what's required to our internal handling of the processed config (i.e. they should model what it should look like after your defaults changes have been applied). That means that `ssl-context-factory` could be called without having the `allow-renegotiation` key set. This isn't a huge deal since `nil` is returned by destructuring an unset key in a map [here](https://github.com/puppetlabs/trapperkeeper-webserver-jetty9/pull/186/files#diff-1fc7ef9acd5fbc52947b0a4ba167e727R191) and `nil` will count as a `false`-y value when we check [further down](https://github.com/puppetlabs/trapperkeeper-webserver-jetty9/pull/186/files#diff-1fc7ef9acd5fbc52947b0a4ba167e727R216).

Functionally, this creates the same effect, so I'm not going to stress it for now.